flocker/control/script.py

```diff
@@ -77,6 +81,35 @@ def main(self, reactor, options):
 
 
 def flocker_control_main():
+    pr = cProfile.Profile()
+
+    def enable_profiling(signal, frame):
+        """
+        Enable profiling of the control service.
+
+        :param int signal: See ``signal.signal``.
+        :param frame: None or frame object. See ``signal.signal``.
+        """
+        pr.enable()
+
+    def output_profile(signal, frame):
+        """
+        Dump profiling statistics to a file.
+        This disables profiling.
+
+        :param int signal: See ``signal.signal``.
+        :param frame: None or frame object. See ``signal.signal``.
+        """
+        current_time = time.strftime("%Y%m%d%H%M%S")
+        path = FilePath('/etc/flocker/profile-{}'.format(current_time))
```

Comment:
`/etc` is the wrong place to put this file.  Either `/var/...` or `/tmp` would be better.